src/server/shell/decoration/basic_decoration.h

```diff
@@ -75,25 +74,27 @@ class BasicDecoration
         std::shared_ptr<scene::Surface> const& window_surface);
     ~BasicDecoration();
 
-    void window_state_updated();
-    void input_state_updated();
+    void redraw();
```

Comment:
`redraw()` can probably be made private - it is only accessed by member functions.